docs/dev/table/modules.md

```diff
@@ -0,0 +1,128 @@
+---
+title: "Modules"
+is_beta: true
+nav-parent_id: tableapi
+nav-pos: 90
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+Modules can extend Flink's built-in objects by defining a set of metadata, for example, functions.
+
+Modules are pluggable. Flink provides a few pre-built modules. Users can write their own too.
+
+For example, users can define their own geo functions and plug them into Flink as built-in functions to be used in 
+Flink SQL and Table APIs. Another example is users can load an out-of-shelf Hive module to use Hive built-in 
+functions as part of Flink built-in functions.
+
+* This will be replaced by the TOC
+{:toc}
+
+## Module Types
+
+### CoreModule
+
+`CoreModule` currently contains all system (built-in) functions. It's loaded by default.
```

Comment:
`CoreModule` contains all of Flink's system (built-in) functions and is loaded by default.